index.js

```diff
@@ -336,32 +336,28 @@ Notifications._onNotification = function(data, isFromBackground = null) {
 
   if ( this.onNotification !== false ) {
     if ( Platform.OS === 'ios' ) {
+      const notifData = data.getData();
       this.onNotification({
-        id: data?.userInfo?.id,
+        id: notifData?.id,
         foreground: ! isFromBackground,
         userInteraction: isFromBackground,
         message: data.getMessage(),
-        data: data.getData(),
+        data: notifData,
         badge: data.getBadgeCount(),
-        alert: data.getAlert(),
-        sound: data.getSound(),
-        fireDate: data._fireDate,
+        title: data.getTitle(),
+        soundName: data.getSound(),
+        fireDate: Date.parse(data._fireDate),
         finish: (res) => data.finish(res)
       });
     } else {
       var notificationData = {
         foreground: ! isFromBackground,
         finish: () => {},
-        ...data
+        ...data,
+        data: data?.userInfo,
```

Comment:
Hi @lukebars 
I'm not sure on this line, I will test on local but this is probably overriding the data field of FCM.